### PR TITLE
fix: return correct exit code when one env failed

### LINKED ISSUE #16: Return RunFailed error
### Describe this problem

When cases fail to run, sqlness return `0`as the result which means success.
`run_env` needs to return errors up instead of logging.
Refer to https://github.com/CeresDB/sqlness/blob/20db562fdfbbd5814fb58a811b52b2bd31a9be63/src/runner.rs#L80.

### Steps to reproduce

1. Define a wrong case, run sqlness.
2. `echo $?` will return 0 while the correct result should be 1.

### Expected behavior

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by jiacai2050**
# Which issue does this PR close?

Closes #16

# Rationale for this change
 
When one env run failed, its error will be ignored, which is not what we expect.

# What changes are included in this PR?

- Return error out when a env run failed
- Add a bad example to test this.
  - It will also write a file lock as liveness flag, CI will ensure it's got deleted after running
-  Remove some unnecessary code.

# Are there any user-facing changes?

No

# How does this change test

New bad example

### CODE REVIEW

File: examples/basic.rs [hunk 1]
```diff
@@ -44,15 +44,12 @@ impl EnvController for MyController {
 
 #[tokio::main]
 async fn main() {
-    let args: Vec<String> = env::args().collect();
```
Comment (jiacai2050) [inline examples/basic.rs:47]: Remove unnecessary part, the basic example should be as simple as possible.

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 /target
+examples/bad-case/simple/select.output
```
Comment (waynexia) [inline .gitignore:5]: Not a good practice to ignore `.output`. Maybe add a `rm` in makefile?

File: src/runner.rs [hunk 3]
```diff
@@ -78,8 +77,11 @@ impl<E: EnvController> Runner<E> {
             };
             let db = self.env_controller.start(&env, config_path).await;
             if let Err(e) = self.run_env(&env, &db).await {
-                println!("Environment {} run failed with error {:?}", env, e);
+                println!("Environment {} run failed, error:{:?}.", env, e);
+                self.env_controller.stop(&env, db).await;
+                return Err(e);
```
Comment (waynexia) [inline src/runner.rs]: Error handling here should also consider the `fail_fast` option. If not fail fast, it should keep running the next env and collect errors after all then return.

File: examples/basic.rs (hunk 1 above)
Comment (waynexia) [reply examples/basic.rs:47]: 👍

File: .gitignore (hunk 2 above)
Comment (jiacai2050) [reply .gitignore:5]: Not for regular case, but in this case, this example will always fail, if we delete this output file, then how will user `diff *result *output`?

File: src/runner.rs (hunk 3 above)
Comment (jiacai2050) [reply src/runner.rs]: Make sense, will fix this.

File: .gitignore (hunk 2 above)
Comment (waynexia) [reply .gitignore:5]: Got it. How about adding a comment here (in `.gitignore`, I'm not sure if there is a better place to specify this rule)

Comment (jiacai2050) [reply .gitignore:5]: Added

Comment (waynexia) [reply .gitignore:5]: 👍

**Review by waynexia — APPROVED** [review]

**Pull request merged**